sagemaker-pipeline-compare-model-versions/notebook.ipynb

```diff
@@ -0,0 +1,703 @@
+    "\n",
+    "![Compare versions](images/compare-versions.png \"Compare versions\")\n",
+    "\n",
+    "\n",
+    "### MLOps\n",
+    "\n",
+    "MLOps, or Machine Learning Operations, is the concept of applying DevOps practices on the lifecycle of a machine learning model. It's the process of creating automated workflows that enable organizations to iterate on machine learning models after they have been deployed into a production environment. MLOps usually consists of two workflows that sits on either side of a machine learning model registry; one to train a model and one to deploy a model. A model registry is a central location to manage machine learning models, where ML engineers and data scientists can compare different model versions, visualize metrics, and decide which versions to accept and which to reject. Ideally, approving a new version of a model triggers a pipeline that ultimately deploys the model into production.\n",
+    "\n",
+    "\n",
+    "From a high-level perspective, it can look like this.\n",
+    "\n",
+    "![High-level MLOps](images/high-level.png \"MLOps from a high-level\")\n",
+    "\n",
+    "The deployment pipeline is what is most similar to traditional CI/CD methods, propagating an artifact through a pipeline that performs testing and if the tests are successful, deploys them. Any CI/CD tools can be used for the deployment, but to automate the entire machine learning model lifecycle, a pipeline on \"the other side\" of the model registry is required as well. Another reusable pipeline through which new data can be pushed when monitoring identifies some form of data or model drift. A pipeline that ultimately produces a new machine learning model and registers it with the model registry. A SageMaker Pipeline.\n",
+    "\n",
+    "![High-level MLOps with training pipeline](images/high-level-train.png \"High-level MLOps with training pipeline\")\n",
+    "\n",
+    "\n",
+    "**This notebook**, demonstrate how SageMaker Pipelines can be used to create a reusable machine learning pipeline that preprocesses, trains, evaluates and registers a machine learning model with the SageMaker Model Registry for visualization and comparison of different model versions. \n",
+    "\n",
+    "### Amazon SageMaker Pipelines\n",
+    "\n",
+    "Amazon SageMaker Pipelines is a purpose-built, easy-to-use CI/CD service for machine learning. With SageMaker Pipelines, customers can create machine learning workflows with an easy-to-use Python SDK, and then visualize and manage workflows using Amazon SageMaker Studio.\n",
+    "\n",
+    "#### SageMaker Pipeline steps\n",
```

Comment:
if you'd like to make this notebook more concise, you can drop all the explanation of steps as its pretty much replicating everything which is mentioned in the docs here: https://docs.aws.amazon.com/sagemaker/latest/dg/pipelines-sdk.html. instead you can just write:

"SageMaker pipelines works on the concept of steps. The order steps are executed in is inferred from the dependencies each step have. If a step has a dependency on the output from a previous step, it's not executed until after that step has completed successfully. To know more about the type of steps and parameters supported, check out the #SageMaker Pipelines Overview(https://docs.aws.amazon.com/sagemaker/latest/dg/pipelines-sdk.html)"